inc/sitemaps/class-author-sitemap-provider.php

```diff
@@ -41,43 +55,31 @@ public function get_index_links( $max_entries ) {
 			return [];
 		}
 
-		// @todo Consider doing this less often / when necessary. R.
-		$this->update_user_meta();
-
-		$has_exclude_filter = has_filter( 'wpseo_sitemap_exclude_author' );
-
-		$query_arguments = [];
+		$query = $this->repository
+			->query()
+			->select_many( 'id', 'permalink', 'object_last_modified' )
+			->where( 'object_type', 'user' )
+			->where_raw( '( is_robots_noindex = 0 OR is_robots_noindex IS NULL )' )
+			->order_by_desc( 'object_last_modified' );
 
-		if ( ! $has_exclude_filter ) { // We only need full users if legacy filter(s) hooked to exclusion logic. R.
-			$query_arguments['fields'] = 'ID';
+		$users_to_exclude = $this->exclude_users( [] );
+		if ( count( $users_to_exclude ) > 0 ) {
+			$query->where_not_in( 'object_id', $users_to_exclude );
 		}
 
-		$users = $this->get_users( $query_arguments );
-
-		if ( $has_exclude_filter ) {
-			$users = $this->exclude_users( $users );
-			$users = wp_list_pluck( $users, 'ID' );
-		}
-
-		if ( empty( $users ) ) {
-			return [];
-		}
-
-		$index      = [];
-		$page       = 1;
-		$user_pages = array_chunk( $users, $max_entries );
+		$indexables = $query->find_many();
+		$user_pages = array_chunk( $indexables, $max_entries );
 
 		if ( count( $user_pages ) === 1 ) {
 			$page = '';
 		}
 
 		foreach ( $user_pages as $users_page ) {
+			array_shift( $users_page );
```

Comment:
@jdevalk this line should be removed, right? It makes it that the second most recent last modified date is shown for each page instead of the most recent. Or is there anything I'm missing?